testing/trino-testing/src/main/java/io/trino/testing/TestingConnectorBehavior.java

```diff
@@ -67,6 +67,7 @@
     SUPPORTS_RENAME_MATERIALIZED_VIEW_ACROSS_SCHEMAS(SUPPORTS_RENAME_MATERIALIZED_VIEW),
 
     SUPPORTS_INSERT,
+    SUPPORTS_INSERT_NOT_NULL_COLUMN(SUPPORTS_INSERT),
```

Comment:
call it `SUPPORTS_NOT_NULL_CONSTRAINT`

Reply:
No, that's misleading (and the cause for my earlier confusion). The constraint is applied by the engine and all connectors "support" it. This is about more about whether the connector allows missing columns on insert (so that remote database replaces missing columns with default values).

Maybe `SUPPORTS_DEFAULT_VALUES_FOR_MISSING_COLUMNS` or `SUPPORTS_MISSING_COLUMNS_ON_INSERT`?

Reply:
>  The constraint is applied by the engine and all connectors "support" it

it is applied by the engine, if the connector declares the column as NOT NULL.
It may fail to update `io.trino.spi.connector.ColumnMetadata#nullable`.

Also, in case `io.trino.spi.connector.ConnectorMetadata#supportsMissingColumnsOnInsert`, the engine does not enforce NOT NULLs, right?

> This is about more about whether the connector allows missing columns on insert

This should be invisible from end user perspective, so we shouldn't ned a `TestingConnectorBehavior` for this.

> so that remote database replaces missing columns with default values).

isn't this related to (covered by) `io.trino.testing.AbstractTestDistributedQueries#testInsertForDefaultColumn`?

Reply:
@hashhar wdyt?

Reply:
I re-read both the tests (testInsertForDefaultColumn and the one we moved here). It's slightly confusing that missing columns on insert is tested as part of `testInsertIntoNotNullColumn`. Maybe we should remove that (since it's covered by `testInsertForDefaultColumn` already).

Agreed on all other points.

cc: @ebyhr

Reply:
> It's slightly confusing that missing columns on insert is tested as part of testInsertIntoNotNullColumn.

i may agree.
It's important to test INSERT with NOT NULL with explicit values _and_ defaults, but whether this is single test method, or two test methods is secondary.

> Maybe we should remove that (since it's covered by testInsertForDefaultColumn already).

or, maybe we let `testInsertForDefaultColumn` test aspects of INSERTS other than NOT NULL constraints?

Reply:
I agree. I'll send a PR that applies your comments.

Reply:
Seems @ebyhr resolved this already in https://github.com/trinodb/trino/pull/10278